packages/preview2-shim/lib/io/worker-io.js

```diff
@@ -0,0 +1,205 @@
+   */
+  static _create (streamType, createPayload) {
+    const stream = new InputStream();
+    stream.#id = ioCall(calls.INPUT_STREAM_CREATE | streamType, null, createPayload);
+    stream.#streamType = streamType;
+    return stream;
+  }
+}
+
+export const _inputStreamCreate = InputStream._create;
+delete InputStream._create;
+
+class OutputStream {
+  #id;
+  #streamType;
+  get _id () {
+    return this.#id;
+  }
+  checkWrite(len) {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_CHECK_WRITE | this.#streamType, this.#id, len);
+  }
+  write(buf) {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_WRITE | this.#streamType, this.#id, buf);
+  }
+  blockingWriteAndFlush(buf) {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_BLOCKING_WRITE_AND_FLUSH | this.#streamType, this.#id, buf);
+  }
+  flush() {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_FLUSH | this.#streamType, this.#id);
+  }
+  blockingFlush() {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_BLOCKING_FLUSH | this.#streamType, this.#id);
+  }
+  writeZeroes(len) {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_WRITE_ZEROES | this.#streamType, this.#id, len);
+  }
+  blockingWriteZeroesAndFlush(len) {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_BLOCKING_WRITE_ZEROES_AND_FLUSH | this.#streamType, this.#id, len);
+  }
+  splice(src, len) {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_SPLICE | this.#streamType, this.#id, src.#id, len);
+  }
+  blockingSplice(src, len) {
+    return streamIoErrorCall(calls.OUTPUT_STREAM_BLOCKING_SPLICE | this.#streamType, this.#id, src.#id, len);
+  }
+  subscribe() {
+    return pollableCreate(ioCall(calls.OUTPUT_STREAM_SUBSCRIBE | this.#streamType, this.#id));
+  }
+  [symbolDispose]() {
+    ioCall(calls.OUTPUT_STREAM_DISPOSE, this.#id);
+  }
+
+  /**
+   * @param {OutputStreamType} streamType
+   * @param {any} createPayload
+   */
+  static _create (streamType, createPayload) {
+    const stream = new OutputStream();
+    stream.#id = ioCall(calls.OUTPUT_STREAM_CREATE | streamType, null, createPayload);
+    stream.#streamType = streamType;
+    return stream;
+  }
+}
+
+export const _outputStreamCreate = OutputStream._create;
+delete OutputStream._create;
+
+export const streams = { Error: IoError, InputStream, OutputStream };
+
+class Pollable {
+  #id;
+  #ready = false;
+  ready () {
+    if (this.#ready)
+      return true;
+    const ready = ioCall(calls.POLL_POLLABLE_READY, this.#id);
+    if (ready)
+      this.#ready = true;
+    return ready;
+  }
+  block () {
+    if (!this.#ready) {
+      ioCall(calls.POLL_POLLABLE_BLOCK, this.#id);
+      this.#ready = true;
+    }
+  }
+  static _create (id) {
+    const pollable = new Pollable();
+    pollable.#id = id;
+    return pollable;
+  }
+  static _listToIds (list) {
+    return list.map(pollable => pollable.#id);
+  }
+  static _markReady (pollable) {
+    pollable.#ready = true;
+  }
+}
+
+const pollableCreate = Pollable._create;
+delete Pollable._create;
```

Comment:
> `delete Pollable._create;`
I've seen this pattern across different files. Can you provide a short explanation (for my learning purpose)?

Reply:
`Pollable` is exposed publicly to users, but users can't create their own `Pollable`s. So the create function is a private function, hence we delete it to not give users access. The reason it has to exist as a class method is because it accesses the private fields.